### PR TITLE
Add OutOfRangeError and handle it properly

### PR DESCRIPTION
**Pull request by epapbak**
# Description

Follow up of #582, introducing proper error type and handling

## Type of change

- New feature (non-breaking change which adds functionality)

## Testing steps

CI

## Checklist
* [x] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [x] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

**Review by JiriPapousek — APPROVED** [review]

File: types/conversion_test.go [hunk 1]
```diff
@@ -38,8 +39,15 @@ func TestUint64ToUint32(t *testing.T) {
 				t.Errorf("Uint64ToUint32() error = %v, wantErr %v", err, tt.wantErr)
 				return
 			}
-			if got != tt.want {
-				t.Errorf("Uint64ToUint32() = %v, want %v", got, tt.want)
+			if tt.wantErr {
+				// Check if the error is of type OutOfRangeError
+				if _, ok := err.(*OutOfRangeError); !ok {
```
Comment (juandspy) [inline types/conversion_test.go:44]: shouldn't you use `tt.errType`?

File: types/conversion_test.go [hunk 2]
```diff
@@ -38,8 +39,15 @@ func TestUint64ToUint32(t *testing.T) {
...
+					t.Errorf("Expected error type *OutOfRangeError, got %T", err)
```
Comment (juandspy) [inline types/conversion_test.go:45]: same here

File: types/conversion_test.go (hunk 1 above)
Comment (epapbak) [reply types/conversion_test.go:44]: I can =)

File: types/conversion_test.go (hunk 2 above)
Comment (juandspy) [reply types/conversion_test.go:42]: I meant `} else if _, ok := err.(*tt.errType); !ok {` 🙈

Comment (juandspy) [reply types/conversion_test.go:42]: and `t.Errorf("Expected error type %v, got %T", tt.errType,  err)`

**Review by juandspy — APPROVED** [review]

**Pull request merged**